mlir/lib/Dialect/SPIRV/IR/SPIRVCanonicalization.cpp

```diff
@@ -115,6 +115,196 @@ void spirv::AccessChainOp::getCanonicalizationPatterns(
   results.add<CombineChainedAccessChain>(context);
 }
 
+//===----------------------------------------------------------------------===//
+// spirv.IAddCarry
+//===----------------------------------------------------------------------===//
+
+// We are required to use CompositeConstructOp to create a constant struct as
+// they are not yet implemented as constant, hence we can not do so in a fold.
+struct IAddCarryFold final : OpRewritePattern<spirv::IAddCarryOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(spirv::IAddCarryOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    auto operands = op.getOperands();
```

Comment:
We can do:
```c++
Value lhs = op.getOperand1();
Value rhs = op.getOperand2();
```

If this is not a very intuitive naming, I can add extra getters `op.getLhs()` and `op.getRhs()` -- WDYT?

Reply:
Agree this looks better. I think it is okay to be done locally rather than having the getters.